packages/dataviews/src/field-types/datetime.tsx

```diff
@@ -0,0 +1,96 @@
+/**
+ * WordPress dependencies
+ */
+import { BaseControl, TimePicker, SelectControl } from '@wordpress/components';
+import { useCallback } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import { useInstanceId } from '@wordpress/compose';
+
+/**
+ * Internal dependencies
+ */
+import type {
+	SortDirection,
+	ValidationContext,
+	DataFormControlProps,
+} from '../types';
+
+function sort( a: any, b: any, direction: SortDirection ) {
+	const timeA = new Date( a ).getTime();
+	const timeB = new Date( b ).getTime();
+
+	return direction === 'asc' ? timeA - timeB : timeB - timeA;
+}
+
+function isValid( value: any, context?: ValidationContext ) {
+	if ( context?.elements ) {
+		const validValues = context?.elements.map( ( f ) => f.value );
+		if ( ! validValues.includes( value ) ) {
+			return false;
+		}
+	}
+
+	return true;
+}
+
+function Edit< Item >( {
+	data,
+	field,
+	onChange,
+}: DataFormControlProps< Item > ) {
+	const { id, label } = field;
+	const value = field.getValue( { item: data } );
+
+	const instanceId = useInstanceId( TimePicker, `datetime-field-${ id }` );
+
+	const onChangeControl = useCallback(
+		( newValue: string | null ) =>
+			onChange( ( prevItem: Item ) => ( {
+				...prevItem,
+				[ id ]: newValue,
+			} ) ),
+		[ id, onChange ]
+	);
+
+	if ( field.elements ) {
+		const elements = [
+			/*
+			 * Value can be undefined when:
+			 *
+			 * - the field is not required
+			 * - in bulk editing
+			 *
+			 */
+			{ label: __( 'Select item' ), value: '' },
+			...field.elements,
+		];
+
+		return (
+			<SelectControl
+				label={ label }
+				value={ value }
+				options={ elements }
+				onChange={ onChangeControl }
+				__next40pxDefaultSize
+				__nextHasNoMarginBottom
+			/>
+		);
+	}
+
+	return (
+		<BaseControl id={ instanceId } label={ label }>
+			<TimePicker
+				id={ instanceId }
+				currentTime={ value }
+				onChange={ onChangeControl }
+				hideLabelFromVision
+			/>
+		</BaseControl>
```

Comment:
The visual label "Date" is being associated with the hours input, which is incorrect. If anything, this should be labeled not by a `BaseControl`, but with the fieldset/legend + `BaseControl.VisualLabel` pattern.

As for the click-label-to-focus behavior, see https://github.com/WordPress/gutenberg/pull/64175#discussion_r1706819405. This should be a separate issue.

@jameskoster But then I also have the design-wise question of whether this visual labeling (i.e. having "Date" be immediately followed by a time input) is viable at all.

<img width="727" alt="TimePicker with single 'Date' label" src="https://github.com/user-attachments/assets/4ff70c5c-f542-4439-8cae-82c0cb89e7ac">

If it is, then we could simplify the original `TimePicker` to have that single-label design.

<img width="701" alt="TimePicker with separate Time and Date labels" src="https://github.com/user-attachments/assets/a42fdbd4-e27b-46a3-97c7-534c167a91d5">

It is either necessary to have both visual labels, or just one. I don't think that necessity changes contextually 🤔 

Given that the current DateTimePicker family of components is a mess and needs to be redesigned anyway, what is the best way to get this feature shipped quickly? I'm thinking either:

- Retain both visual labels as in the original TimePicker.
- Add a CSS hack to hide the TimeInput, if that part isn't necessary.
- Change the TimePicker design to have a single visual label.
- If none of the above is viable, then wrap everything in the extra fieldset as described above and make it say something like "Date & Time" — though this is also not ideal because we've been told to avoid nested fieldsets.

Reply:
> Change the TimePicker design to have a single visual label.

For me that's the ideal scenario. Ultimately we're editing one property for us. Whether that means an extra fieldset or a single label for the whole component doesn't really matter much but it should be clear to the user that it's a single property being edited.

Reply:
Sounds good! I'll look into the click-to-focus behavior separately — we have a promising implementation pattern in https://github.com/WordPress/gutenberg/pull/64267#discussion_r1707115074.

> Somehow render the field name as a "label/legend" in `datetime.tsx` (where the current `BaseControl` is)

I would recommend going with the fieldset + legend as `BaseControl.VisualLabel` pattern for now. On my end (not a blocker for this PR), I'll see if I can remove some unnecessary fieldsets in TimePicker itself, using the labeling strategy from [React Aria](https://react-spectrum.adobe.com/react-aria/TimeField.html).

Reply:
> Change the TimePicker design to have a single visual label.

If this is the ideal scenario doesn't that point us towards a design for `TimePicker` that is more like the [`datetime-local` input type](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/datetime-local)?

Reply:
@mirka in https://github.com/WordPress/gutenberg/pull/64267/commits/fbafe37b82cc2a3dc1de991eb269318afd4d98ae I pushed the legend+visuallabel approach. One thing I'm uneasy about is that it requires me to export the styled `Fieldset` component as part of the component. Let me know what you think of this or if you had a something different in mind. Feel free to push directly to this PR if it's faster/more convenient to explain your point.

Reply:
Oh sorry no, I just meant to use the `fieldset` HTML element. We don't officially provide a capital-F `Fieldset` component with style resets, so you can just add your own ad hoc reset styles with a CSS class.

Some of the styles in here aren't even appropriate for this use case, so let's just add whatever is minimally required.

https://github.com/WordPress/gutenberg/blob/4ea9b9f0dd92c1f3a58e5eb25f1d26a65871fe13/packages/components/src/date-time/time/styles.ts#L23-L31

Reply:
> If this is the ideal scenario doesn't that point us towards a design for `TimePicker` that is more like the [`datetime-local` input type](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/datetime-local)?

Maybe? I wasn't around when this component was introduced, but it currently isn't satisfying our composition needs at this point. It's definitely time for a v2.

Reply:
> Oh sorry no, I just meant to use the fieldset HTML element. We don't officially provide a capital-F Fieldset component with style resets, so you can just add your own ad hoc reset styles with a CSS class.

Nice, thanks! I pushed this at https://github.com/WordPress/gutenberg/pull/64267/commits/02f1e35fe1782bbcf15e36824ce4c48cea1caf78